pkg/server/structlogging/hot_ranges_log.go

```diff
@@ -48,18 +57,42 @@ var TelemetryHotRangesStatsLoggingDelay = settings.RegisterDurationSetting(
 	1*time.Second,
 )
 
+// TelemetryHotRangesStatsCPUThreshold defines the cpu duration
+// per second which needs to be exceeded for the system to automatically
+// log the hot ranges. It tracks the reasoning that the kv layer
+// uses to determine when to begin sampling reads for a given
+// range in the keyspace, more information found where the cluster
+// setting SplitByLoadCPUThreshold is defined.
+var TelemetryHotRangesStatsCPUThreshold = settings.RegisterDurationSetting(
+	settings.SystemOnly,
+	"server.telemetry.hot_ranges_stats.cpu_threshold",
+	"the cpu time over which the system will automatically begin logging hot ranges",
+	time.Second/4,
+)
+
+type HotRangesServer interface {
```

Comment:
what's the purpose of this? where is it being used?

I'd presume its for tests, if so can we move it to the test file?